src/app/modules/submission/email-submission/email-submission.middleware.ts

```diff
@@ -38,20 +43,30 @@ export const prepareEmailSubmission: RequestHandler<
   const hashedFields =
     (res as ResWithHashedFields<typeof res>).locals.hashedFields || new Set()
   let emailData: EmailData
+  // TODO (#847): remove when we are sure of the shape of responses
   try {
     emailData = EmailSubmissionService.createEmailData(
       req.body.parsedResponses,
       hashedFields,
     )
   } catch (error) {
     logger.error({
-      message: 'Failed to create answer template',
+      message: 'Failed to create email data',
       meta: {
-        action: 'getFormattedResponse',
-        questions: req.body.parsedResponses.map(
-          (response) => response?.question,
-        ),
-        keys: req.body.parsedResponses.map(Object.keys),
+        action: 'prepareEmailSubmission',
+        responseMetaData: req.body.parsedResponses.map((response) => ({
+          question: response?.question,
+          answerTruthy:
+            response.fieldType !== BasicField.Table &&
+            response.fieldType !== BasicField.Checkbox &&
+            !!response?.answer,
+          answerArrayTruthy:
+            (response.fieldType === BasicField.Table ||
+              response.fieldType === BasicField.Checkbox) &&
+            !!response?.answerArray,
+          _id: response?._id,
+          fieldType: response?.fieldType,
```

Comment:
would it be better not to make assumptions about whether the truthiness of those keys is related to the table and checkbox fields?

Reply:
true, tbh I did it this way just to make the types line up - because our `ProcessedResponse` types are quite strong, we're only allowed to access `answerArray` for `Table` and `Checkbox` and `answer` for everything other than those two fieldtypes. but in this case it might be better to just cast the response, since clearly the types aren't behaving as expected.

Reply:
also do you think it will be more helpful to log `typeof response?.answer` and `Array.isArray(response?.answerArray)`?

Reply:
sure! I would think that our Joi validation + the tightened field validation would have already taken care of that, but at this point I'm questioning all our basic assumptions lol